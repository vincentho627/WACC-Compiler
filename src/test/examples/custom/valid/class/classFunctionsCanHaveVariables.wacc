# Class Functions can have variables


begin
  class Vehicle is
    private string name ;
    public int tyres 
    public string getName() is 
      return name
    end
    public int getNumber() is
      return tyres
    end
    public int setNumber() is
      int offset = 5 ;
      tyres = tyres * offset ;
      return tyres
    end
  end
  Vehicle bus = new Vehicle("Bus", 4) ;
  int number1 = call bus.getNumber() ;
  println number1 ;
  int number2 = call bus.setNumber() ;
  println number2 ;
  int number3 = call bus.getNumber() ;
  println number3 ;
  bus.tyres = 5 ;
  println bus.tyres ;
  int number4 = call bus.getNumber() ;
  println number4 
end








