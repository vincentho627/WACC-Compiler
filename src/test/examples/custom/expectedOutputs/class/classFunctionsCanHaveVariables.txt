4
20
20
5
5
